k8s-scheduler/src/test/java/com/vmware/dcm/ScopeTest.java

```diff
@@ -0,0 +1,263 @@
+import io.fabric8.kubernetes.api.model.NodeStatus;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.PodSpec;
+import io.fabric8.kubernetes.api.model.PodStatus;
+import io.fabric8.kubernetes.api.model.Quantity;
+import io.fabric8.kubernetes.api.model.ResourceRequirements;
+import org.jooq.DSLContext;
+import org.jooq.Result;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Tests for Scope
+ */
+@ExtendWith({})
+public class ScopeTest {
+    private static final int NUM_THREADS = 1;
+
+    @Test
+    public void testSpareCapacityFilter() {
+        final DBConnectionPool dbConnectionPool = new DBConnectionPool();
+        final DSLContext conn = dbConnectionPool.getConnectionToDb();
+
+        final NodeResourceEventHandler nodeResourceEventHandler = new NodeResourceEventHandler(dbConnectionPool);
+        final PodEventsToDatabase eventHandler = new PodEventsToDatabase(dbConnectionPool);
+        final PodResourceEventHandler handler = new PodResourceEventHandler(eventHandler::handle);
+
+        final int numNodes = 10;
+        final int numPods = 4;
+
+        for (int i = 0; i < numNodes; i++) {
+            final Node node = newNode("n" + i, Collections.emptyMap(), Collections.emptyList());
+            // more spare capacity as i increases
+            node.getStatus().getCapacity().put("cpu", new Quantity(String.valueOf(10 + i)));
+            node.getStatus().getCapacity().put("memory", new Quantity(String.valueOf(1000)));
+            node.getStatus().getCapacity().put("pods", new Quantity(String.valueOf(100)));
+            nodeResourceEventHandler.onAddSync(node);
+        }
+        for (int i = 0; i < numPods; i++) {
+            handler.onAddSync(newPod("p" + i));
+        }
+
+        Model model = buildModel(conn);
+        ScopedModel scopedModel = new ScopedModel(conn, model);
+        Set<String> scopedNodes = scopedModel.getScopedNodes();
+
+        assertEquals(numPods, scopedNodes.size());
+        scopedNodes.forEach(n -> assertTrue(Integer.parseInt(n.substring(1)) >= numNodes - numPods));
+    }
+
+    /*
+     * E2E test with scheduler:
+     * Test if Scope limits candidate nodes according to spare resources
+     */
+    @Test
+    public void testScopedSchedulerSimple() {
+        final DBConnectionPool dbConnectionPool = new DBConnectionPool();
+        final DSLContext conn = dbConnectionPool.getConnectionToDb();
+        final List<String> policies = Policies.getDefaultPolicies();
+        final NodeResourceEventHandler nodeResourceEventHandler = new NodeResourceEventHandler(dbConnectionPool);
+        final int numNodes = 100;
+        final int numPods = 40;
+        conn.insertInto(Tables.BATCH_SIZE).values(numPods).execute();
```

Comment:
You can remove this line. It's a vestige of an older implementation and is not actually used.